### ISSUE TITLE
improve na_ofi error messages when libfabric providers aren't available

### ISSUE THREAD

**Issue by carns** [mercury-hpc/mercury#395]
**Is your feature request related to a problem? Please describe.**

When initializing Mercury with OFI, there are generally three places where the providers need to be in alignment: the string provided by the user (i.e. "tcp://"), the set of providers translated by mercury (i.e. "tcp;ofi_rxm"), and the set of providers supported by the libfabric library being used (i.e. "rxm,tcp,sockets,verbs").  If any of these are not in alignment then the error message can be confusing.  The user might not realize that Mercury translated the requested provider, and it isn't immediately obvious that the libfabric build doesn't support the requested combination.  It is usually manifested as a numeric error code from libfabric and a fairly deep Mercury error stack.

**Describe the solution you'd like**

A clear error message from Mercury when OFI fails to initialize that says something like "OFI library does not support the requested provider.  The user requested "tcp", Mercury translated this to "tcp;ofi_rxm", but the libfabric build only supports "tcp,verbs,sockets".  Please link against a libfabric library that supports both "tcp" and "rxm".

We would only attempt to assess this situation if na_ofi fails to open the endpoint.  Then we still only display a message if we detect a provider mismatch.  Otherwise we fall back to the usual existing error messages.

The above suggestion assumes that there is a way to query what providers are available in libfabric at runtime.

**Describe alternatives you've considered**

**Additional context**

**Comment by carns** [comment]
Just commenting here for reference, the fi_info utility in libfabric has an example of how to iterate through the results of an fi_getinfo() call and print the provider names:

https://github.com/ofiwg/libfabric/blob/master/util/info.c#L268

**Comment by carns** [comment]
To reproduce:

```
[carns@carns-x1-7g bin (master)]$ ./hg_test_server -c ofi -p tcp
# Using info string: ofi+tcp://
# Starting server with 8 threads...
# Waiting for client

[carns@carns-x1-7g bin (master)]$ ./hg_test_server -c ofi -p verbs
# Using info string: ofi+verbs://
# NA -- Error -- /home/carns/working/src/mochi/mercury/src/na/na_ofi.c:1458
 # na_ofi_getinfo(): fi_getinfo() failed, rc: -61 (No data available)
# NA -- Error -- /home/carns/working/src/mochi/mercury/src/na/na_ofi.c:2959
 # na_ofi_check_protocol(): na_ofi_getinfo() failed
# NA -- Error -- /home/carns/working/src/mochi/mercury/src/na/na.c:277
 # NA_Initialize_opt(): Specified class name does not support requested protocol
# NA -- Error -- /home/carns/working/src/mochi/mercury/Testing/na/na_test.c:429
 # NA_Test_init(): Could not initialize NA
# HG Test -- Error -- /home/carns/working/src/mochi/mercury/Testing/mercury_test.c:364
 # HG_Test_init(): NA_Test_init() failed (NA_PROTOCOL_ERROR)
# HG Test -- Error -- /home/carns/working/src/mochi/mercury/Testing/test_server.c:135
 # main(): HG_Test_init() failed
```

The second example above attempts to start the verbs provider if libfabric has not been compiled with verbs support.  The error message could be clearer in this case.

**Comment by carns** [comment]
Addressed in #396
